Source/misdat.cpp

```diff
@@ -127,68 +129,120 @@ MissileDataStruct MissileData[] = {
+	{ "reflect",  MFILE_REFLECT,    1,       MFLAG_LOCK_ANIMATION,    0,             1,                                    160,                     64               },
+	{ "ms_ora",   MFILE_LICH,      16,       MFLAG_HIDDEN,            0,            15,                                     96,                      8               },
+	{ "ms_bla",   MFILE_MSBLA,     16,       MFLAG_HIDDEN,            0,            15,                                     96,                      8               },
+	{ "ms_reb",   MFILE_NECROMORB, 16,       MFLAG_HIDDEN,            0,            15,                                     96,                      8               },
+	{ "ms_yeb",   MFILE_ARCHLICH,  16,       MFLAG_HIDDEN,            0,            15,                                     96,                      8               },
+	{ "rglows1",  MFILE_RUNE,       1,       0,                       0,            10,                                     96,                      8               },
+	{ "ex_yel2",  MFILE_EXYEL2,     1,       MFLAG_HIDDEN,            0,            10,                                    220,                     78               },
+	{ "ex_blu2",  MFILE_EXBL2,      1,       MFLAG_HIDDEN,            0,            10,                                    212,                     86               },
+	{ "ex_red3",  MFILE_EXRED3,     1,       MFLAG_HIDDEN,            0,             7,                                    292,                    114               },
+	{ "ms_blb",   MFILE_BONEDEMON, 16,       MFLAG_HIDDEN,            0,            15,                                     96,                      8               },
+	{ "ex_ora1",  MFILE_EXORA1,     1,       MFLAG_HIDDEN,            0,            13,                                     96,                    -12               },
+	{ "ex_blu3",  MFILE_EXBL3,      1,       MFLAG_HIDDEN,            0,             7,                                    292,                    114               },
+	{ "",         MFILE_NONE,       0,       0                                                                                                                       },
 	// clang-format on
 };
 
+MisFileData::MisFileData(const char *name, uint8_t animName, uint8_t animFAmt, uint32_t flags,
+			 AutofillArray<uint8_t, 16> animDelay, AutofillArray<uint8_t, 16> animLen,
+			 AutofillArray<int16_t, 16> animWidth, AutofillArray<int16_t, 16> animWidth2)
+	: name(name)
+	, animName(animName)
+	, animFAmt(animFAmt)
+	, flags(flags)
+	, animDelay(animDelay)
+	, animLen(animLen)
+	, animWidth(animWidth)
+	, animWidth2(animWidth2)
+{
+	if (flags & MFLAG_ALLOW_SPECIAL)
+		pinnedMem.reserve(1);
+	else
+		pinnedMem.reserve(animFAmt);
+
+	for (int i = animFAmt; i < 16; i++) {
+		animDelay[i] = 0;
+		animLen[i] = 0;
+		animWidth = 0;
+		animWidth2 = 0;
+	}

```

Comment:
Does the game ever touch entries beyond animFAmt? If not, this is not needed.

Reply:
It does not.